common/src/java/org/apache/hadoop/hive/common/type/Decimal128.java

```diff
@@ -1100,7 +1097,7 @@ public Decimal128 subtractDestructive(Decimal128 right, short scale) {
    */
   public static void multiply(Decimal128 left, Decimal128 right,
       Decimal128 result, short scale) {
-    if (result == left || result == right) {
```

Comment:
I think this is not righ; the exception was intended to obstruct possible bad preference passing; Object.equals is compare by content.

Reply:
Good catch. Will add a test to cover this case and make sure that it is enforced.